### PR TITLE
[CARBONDATA-4000] Presto select query failure right after firing update query.

### PR DESCRIPTION
**Pull request by akkio-97**
### Why is this PR needed?
 The putObject() method implementation was missing due to which select query on **struct** complex data type after **update** queries were failing. 
 
 ### What changes were proposed in this PR?
Added implementation in ColumnarVectorWrapperDirectWithDeleteDelta. Earlier without this implementation control went to its parent class AbstractCarbonColumnarVector,java which was unsupported.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDelta.java [hunk 1]
```diff
@@ -58,6 +58,17 @@ public void putBoolean(int rowId, boolean value) {
     }
   }
 
+  @Override
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDelta.java:61]: please check and handle for `ColumnarVectorWrapperDirectWithInvertedIndex`, `ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex` also

Comment (akkio-97) [reply core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDelta.java:61]: Done. Issue is with regards to only update query so changes may not be required in ColumnarVectorWrapperDirectWithInvertedIndex.

File: core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex.java [hunk 2]
```diff
@@ -78,6 +80,17 @@ public void putNull(int rowId) {
     }
   }
 
+  @Override
+  public void putObject(int rowId, Object obj) {
+    if (!deletedRows.get(rowId)) {
+      if (nullBits.get(rowId)) {
+        columnVector.putNull(counter++);
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex.java:87]: can you locally test one table with inverted index, I think here should not use `counter`, may be ned to use `rowId`, convert function is having `counter` already

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2408/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4149/

**Comment by ajantha-bhat** [comment]
LGTM. 
Will merge once build passes.

Please create jira for inverted index issue and handle in separate PR

File: core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex.java (hunk 2 above)
Comment (akkio-97) [reply core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex.java]: Have raised a separate jira to address this issue - CARBONDATA-4004

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2440/

**Comment by akkio-97** [comment]
retest this please

**Comment by akkio-97** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4176/

**Comment by akkio-97** [comment]
> LGTM.
> Will merge once build passes.
> 
> Please create jira for inverted index issue and handle in separate PR

Done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2442/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4185/

**Pull request closed without merging**